File: tests/JConditions/jt4.c

jasync localme(int c, char *s) {
    while(1) {
        jsys.sleep(200000);
        printf("############-->>> Hello  ME  %d... %s\n", c, s);
    }
}

jasync localyou(int c, char *s) {
    while(1) {
        jsys.sleep(100000);
        printf("############-->>> Hello YOU  %d, %s\n", c, s);
        callXAB(s);
    }
}

int main(int argc, char *argv[]) {
    localme(10, "message-from-local-c");
    localyou(10, "message-to-j");
    return 0;
}

File: tests/JConditions/jt4.js


let count = 0;

jcond {
    fogonly(my, your) {
        return my.type === 'fog';
    }
}

jasync {fogonly} callXAB(str: char*) {
    count++;
    console.log(count, str);
}

setInterval(()=> {
    console.log("hello.. main loop ");
}, 1000);
